builtin.c

```diff
@@ -683,8 +683,7 @@ XS(XS_builtin_unimport)
          * tombstone and instead COP_SEQ_MAX_HIGH_set() on the padname to
          * clear it.
          */
-        off = pad_add_name_sv(ampname, padadd_STATE|padadd_TOMBSTONE, 0, 0);
-        SvREFCNT_dec(PL_curpad[off]);
+        pad_add_name_sv(ampname, padadd_STATE|padadd_TOMBSTONE, 0, 0);
```

Comment:
Does the removal of the SvREFCNT_dec() fix some other problem?

If so the commit message should mention it.

I suspect this was resulting in freeing of a freed scalar further down the line but the commit message should mention it if so.

Reply:
Actually it was indirectly *causing* a problem. Nothing ought to be looking in the pad at the SVs in tombstone entires, but if anything did then previously they'd see the address of an SV that has longsince been freed (or perhaps more likely reused). By not freeing it, we instead leave it as a simple SVt_NULL. I thought that was safer than freeing it then NULL'ing it out, in case something tried to dereference it anyway.

Reply:
I've now made some further adjustments to the code in `pad.c`, and expanded on the wording in the commit message to explain the situation.